From eb12d522c5cd2255400f16c6dc74e6c70198bcff Mon Sep 17 00:00:00 2001
From: Ben Chan <benchan@chromium.org>
Date: Mon, 23 Jan 2017 17:56:05 -0800
Subject: [PATCH] Temporarily remove attribute 'deprecated' on g_type_init().

g_type_init() has been deprecated since glib 2.36 and is marked with the
attribute 'deprecated'. Chrome OS currently uses glib 2.34.3 and has a
few call sites of g_type_init() in the code base, which will trigger a
compilation error when compiling against glib 2.36 due to
-Werror=deprecated-declarations. In order to allow a seamless upgrade
from glib 2.34.3 to 2.36.4, this patch temporarily remove the attribute
'deprecated' on g_type_init(). Once the glib is upgraded to 2.36, we
will remove those call sites of g_type_init() and then drop this patch.
---
 gobject/gtype.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gobject/gtype.h b/gobject/gtype.h
index 8a1bff2b4..078f5d7b8 100644
--- a/gobject/gtype.h
+++ b/gobject/gtype.h
@@ -665,7 +665,7 @@ typedef enum	/*< skip >*/
 
 
 /* --- prototypes --- */
-GLIB_DEPRECATED_IN_2_36
+GLIB_AVAILABLE_IN_ALL
 void                  g_type_init                    (void);
 GLIB_DEPRECATED_IN_2_36
 void                  g_type_init_with_debug_flags   (GTypeDebugFlags  debug_flags);
-- 
2.11.0.483.g087da7b7c-goog

